### PR TITLE
[Example] ggml: use wasmedge-wasi-nn crate

### PR DESCRIPTION
**Pull request by dm4**
Because we have replaced the original `wasi-nn` crate with `wasmedge-wasi-nn`, we need to make corresponding modifications in the examples.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: wasmedge-ggml/chatml/Cargo.toml [hunk 1]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/chatml/Cargo.toml:8]: We already published the crate, how about using `0.7.0` directly?

File: wasmedge-ggml/embedding/Cargo.toml [hunk 2]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/embedding/Cargo.toml:8]: Ditto.

File: wasmedge-ggml/gemma/Cargo.toml [hunk 3]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/gemma/Cargo.toml:8]: Ditto.

File: wasmedge-ggml/llama-stream/Cargo.toml [hunk 4]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/llama-stream/Cargo.toml:8]: Ditto.

File: wasmedge-ggml/llama/Cargo.toml [hunk 5]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/llama/Cargo.toml:8]: Ditto.

File: wasmedge-ggml/llava-base64-stream/Cargo.toml [hunk 6]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/llava-base64-stream/Cargo.toml:8]: Ditto.

File: wasmedge-ggml/llava/Cargo.toml [hunk 7]
```diff
@@ -5,4 +5,4 @@ edition = "2021"
 
 [dependencies]
 serde_json = "1.0"
-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
+wasmedge-wasi-nn = { git = "https://github.com/second-state/wasmedge-wasi-nn", branch = "ggml" }
```
Comment (hydai) [inline wasmedge-ggml/llava/Cargo.toml:8]: Ditto.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — APPROVED** [review]

**Pull request merged**